raiden/api/python.py

```diff
@@ -96,9 +101,11 @@ def token_network_register(self, registry_address, token_address):
             # registered with the smart contract and this node has not yet
             # polled for the event (otherwise the check above would have
             # failed).
-            # To provide a consistent view to the user, force an event poll to
-            # register the token network.
-            self.raiden.poll_blockchain_events()
+            #
+            # To provide a consistent view to the user, wait one block, this
+            # will guarantee that the events have been processed.
+            next_block = self.raiden.get_block_number() + 1
+            waiting.wait_for_block(self.raiden, next_block, poll_timeout)
```

Comment:
Waiting for the next block might block quite some time, we should at least mention that in the docstring.